<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Tutorial of UltraGBIF • UltraGBIF</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Tutorial of UltraGBIF">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">UltraGBIF</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">3.0.7</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Tutorial_of_UltraGBIF.html">Tutorial of UltraGBIF</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/wyx619/UltraGBIF/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Tutorial of UltraGBIF</h1>
                        <h4 data-toc-skip class="author">Yuxuan
Wang</h4>
            <address class="author_afil">
      Sichuan
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:wyx1743@gmail.com" class="email">wyx1743@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">2026-01-09</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/wyx619/UltraGBIF/blob/HEAD/vignettes/Tutorial_of_UltraGBIF.Rmd" class="external-link"><code>vignettes/Tutorial_of_UltraGBIF.Rmd</code></a></small>
      <div class="d-none name"><code>Tutorial_of_UltraGBIF.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h2>
<p>This Tutorial parses and merges <a href="https://doi.org/10.15468/dl.bythb4" class="external-link"><em>Saxifraga</em></a>
occurrence records from GBIF.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install UltraGBIF from GitHub. UltraGBIF runs with rWCVPdata,
so install it firstly (We recommend rWCVPdata version 0.6.0 with WCVP
version 14), and the initial installation takes some time.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>repos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"repos"</span><span class="op">)</span>,<span class="st">"https://wyx619.github.io/Repo"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout <span class="op">=</span> <span class="fl">600</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"rWCVPdata"</span><span class="op">)</span> <span class="co">## install rWCVPdata [require long time]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"UltraGBIF"</span><span class="op">)</span> <span class="co">## install UltraGBIF</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="import-records">Import records<a class="anchor" aria-label="anchor" href="#import-records"></a>
</h2>
<p>Initially, download occurrence records from GBIF. Filter records
online such as <em>Basis of record = Preserved specimen, Occurrence
status = present, and Scientific name = Saxifraga</em>. Get the DwC-A
file and remember its absolute path.</p>
<p>This module also utilizes <code>EnumOccurrence</code> Issue values to
transform these issues into a structured format, enabling systematic
quality assessment through individualized issue tracking and
standardized classification. See more <a href="https://gbif.github.io/gbif-api/apidocs/org/gbif/api/vocabulary/OccurrenceIssue.html" class="external-link">details</a>
here.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/wyx619/UltraGBIF" class="external-link">UltraGBIF</a></span><span class="op">)</span></span>
<span><span class="va">gbif_occurrence_file</span> <span class="op">&lt;-</span> <span class="st">"path/to/the/DwC-A file"</span></span>
<span><span class="va">occ_import</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_records.html">import_records</a></span><span class="op">(</span>GBIF_file <span class="op">=</span> <span class="va">gbif_occurrence_file</span>,</span>
<span>                             only_PRESERVED_SPECIMEN <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><code>occ_import</code> includes <code>occ</code> for initial
occurrence records, <code>occ_gbif_issue</code> for GBIF issues and its
<code>summary</code>, and time cost.</p>
</div>
<div class="section level2">
<h2 id="check-taxon-name">Check taxon name<a class="anchor" aria-label="anchor" href="#check-taxon-name"></a>
</h2>
<p>UltraGBIF uses WCVP<span class="citation">(Govaerts et al.
2021)</span> as a high-authority taxonomic backbone, systematically
resolving taxons in GBIF occurrence records. Alternatively, TNRS<span class="citation">(Boyle et al. 2013)</span> is supported by setting the
parameter <code>local_taxon_resolution = FALSE</code>.</p>
<p>To optimize computational efficiency, this module supports
parallelization through multi-threading by specifying the desired
concurrency level via the <code>threads</code> parameter.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">names.checked</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/check_occ_name.html">check_occ_name</a></span><span class="op">(</span>occ_import <span class="op">=</span> <span class="va">occ_import</span>,</span>
<span>                                threads <span class="op">=</span> <span class="fl">4</span>,</span>
<span>                                local_taxon_resolution <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                if_author_fails_try_without_combinations <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><code>names.checked</code> includes <code>occ_wcvp_check_name</code>
for records with resolved taxons and its <code>summary</code>, and time
cost.</p>
</div>
<div class="section level2">
<h2 id="check-collectors-dictionary">Check collectors dictionary<a class="anchor" aria-label="anchor" href="#check-collectors-dictionary"></a>
</h2>
<p>This module extracts the primary collector’s surname from the
<code>recordedBy</code> field in occurrence records and generates an
association list linking standardized surnames to corresponding records.
It assists to resolve collector identity ambiguities and enforce
nomenclatural consistency. Referred <code>CollectorDictionary</code> in
this module comes from ParseGBIF<span class="citation">(De Melo et al.
2024)</span>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dictionary</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_collectors_dictionary.html">prepare_collectors_dictionary</a></span><span class="op">(</span>occ_import <span class="op">=</span> <span class="va">occ_import</span>,</span>
<span>                                            surname_selection_type <span class="op">=</span> <span class="st">"largest_string"</span>,</span>
<span>                                            max_words_name <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                                            min_characters_in_name <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><code>dictionary</code> includes <code>my_dictionary</code> for
checked collectors name and <code>ref_dictionary</code> for referred
dictionary, and time cost.</p>
</div>
<div class="section level2">
<h2 id="generate-unique-collection-mark">Generate unique collection mark<a class="anchor" aria-label="anchor" href="#generate-unique-collection-mark"></a>
</h2>
<p>To resolve duplicate records in biodiversity datasets, UltraGBIF
generates a standardized composite key for each unique collection event
like ParseGBIF<span class="citation">(De Melo et al. 2024)</span> does.
This key is constructed by concatenating three core metadata fields:</p>
<ul>
<li><p><strong>Plant family</strong> (<code>family</code>)</p></li>
<li><p><strong>Standardized collector surname</strong>
(<code>recordedByStandardized</code>)</p></li>
<li><p><strong>Numeric-only collection number</strong>
(<code>recordNumber_Standard</code>)</p></li>
</ul>
<p>The resulting identifier,
<code>key_family_recordedBy_recordNumber</code>, groups specimens
sharing identical collection events. An optional temporal variant,
<code>key_year_recordedBy_recordNumber</code>, incorporates collection
year where available to enhance resolution.</p>
<p>Standardization ensures robust matching: non-numeric characters are
stripped from collection numbers, and collector names are normalized
using predefined dictionaries to reconcile spelling variants.</p>
<p>This module addresses two key challenges:</p>
<ol style="list-style-type: decimal">
<li><p>Disambiguating common collector surnames through taxonomic
context.</p></li>
<li><p>Enabling automated duplicate detection without reliance on
often-missing temporal metadata.</p></li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">collection_key</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_collection_mark.html">generate_collection_mark</a></span><span class="op">(</span>occ_import <span class="op">=</span> <span class="va">occ_import</span>,</span>
<span>                                            dictionary <span class="op">=</span> <span class="va">dictionary</span><span class="op">)</span></span></code></pre></div>
<p><code>collection_key</code> includes
<code>occ_collectorsDictionary</code> for input records updated with
standardized fields and generated keys and
<code>CollectorsDictionary_add</code> for newly encountered collectors
for dictionary expansion and reuseand time costs.</p>
</div>
<div class="section level2">
<h2 id="set-digital-voucher">Set digital voucher<a class="anchor" aria-label="anchor" href="#set-digital-voucher"></a>
</h2>
<p>Following generation of a unique collection event identifier
(collection_key), this module resolves duplicate records and outlines
the selection of the master digital voucher and associated integration
to represent each event.</p>
<p><strong>Duplicate Grouping and Voucher Selection</strong></p>
<ul>
<li><p>For records possessing complete collection event keys, duplicates
are identified and grouped based on identical
<code>collection_key</code> values. Within each group, a master digital
voucher is selected by evaluating a composite score that combines two
dimensions: metadata completeness and geospatial reliability. Metadata
completeness is quantified as the sum of binary presence indicators
across 10 fields, including <code>recordedBy</code>,
<code>locality</code>, and <code>catalogNumber</code>, where presence is
scored as 1 and absence as 0. Geospatial reliability is assigned
according to GBIF’s <code>EnumOccurrenceIssue</code> classification,
with values of −1 indicating unaffected coordinates, −3 indicating
potential positional error, and −9 indicating coordinates unsuitable for
spatial analysis. The record achieving the highest combined score is
designated the master voucher. Any missing metadata in the master record
are supplemented using non-redundant values drawn from other records
within the same group.</p></li>
<li><p>For records with incomplete keys (e.g., missing
<code>recordedBy</code> or <code>recordNumber</code>), no grouping is
performed, and each is treated as a singleton event. Nevertheless,
metadata completeness and geospatial scores are computed identically to
ensure consistent, system-wide data quality assessment.</p></li>
</ul>
<p><strong>Taxonomic Assignment</strong></p>
<ul>
<li><p>For groupable events, the accepted taxon name
(<code>UltraGBIF_sample_taxon_name</code>) is assigned as the most
frequently occurring species- or subspecies-level identification among
duplicates; ties are resolved alphabetically. Unidentified events are
flagged (<code>UltraGBIF_unidentified_sample = TRUE</code>).</p></li>
<li><p>Non-groupable records are processed individually with identical
taxonomic logic.</p></li>
</ul>
<p><strong>Geospatial Integration</strong></p>
<p>If the master voucher has unusable coordinates (score = -9) or lacks
coordinates entirely, usable coordinates from duplicates (score ≥ -3)
are preferentially retained.</p>
<p><strong>Standardized Output Fields</strong></p>
<p>Key output variables include:</p>
<ul>
<li><p><code>UltraGBIF_digital_voucher</code>: identifies master record
(logical);</p></li>
<li><p><code>UltraGBIF_duplicates</code>: presence of duplicates
(logical);</p></li>
<li><p><code>UltraGBIF_num_duplicates</code>: count of grouped
records;</p></li>
<li><p><code>UltraGBIF_non_groupable_duplicates</code>: records with
incomplete keys (logical);</p></li>
<li><p><code>UltraGBIF_duplicates_grouping_status</code>: grouping
feasibility (e.e., “groupable”, “not groupable: no
recordedBy”);</p></li>
<li><p><code>UltraGBIF_sample_taxon_name_status</code>: taxonomic
resolution (“identified”, “divergent”, “not identified”);</p></li>
<li><p><code>UltraGBIF_useful_for_spatial_analysis</code>: coordinate
usability (logical).</p></li>
</ul>
<p>This module ensures robust duplicate resolution, metadata enrichment,
and interoperability with global biodiversity standards.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">voucher</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/set_digital_voucher.html">set_digital_voucher</a></span><span class="op">(</span>occ_import <span class="op">=</span> <span class="va">occ_import</span>,</span>
<span>                               names.checked <span class="op">=</span> <span class="va">names.checked</span>,</span>
<span>                               collection_key <span class="op">=</span> <span class="va">collection_key</span><span class="op">)</span></span></code></pre></div>
<p><code>voucher</code> includes the full flagged digital voucher of
occurrence records and its <code>summary</code>, and time cost.</p>
</div>
<div class="section level2">
<h2 id="refine-records">Refine records<a class="anchor" aria-label="anchor" href="#refine-records"></a>
</h2>
<p>When a <code>voucher</code> is flagged as usable, its missing
information in fields is restored by consolidating data from duplicate
records belonging to identical collection events, creating one clean
record per unique event, whether the event key is complete or not. It
picks the most common species-level name (using alphabetical order to
break ties) or labels the event “unidentified” if no such name exists;
incomplete keys are treated as separate events. Missing coordinates in
the main record are filled in from duplicates.</p>
<p>For finally complete usable vouchers, their associated occurrence
records are automatically subjected to geospatial validation and precise
WGSRPD attribution. Leveraging the R packages CoordinateCleaner<span class="citation">(Zizka et al. 2019)</span> and terra, this module
efficiently performs coordinate validation and cleaning, targeting
common geospatial anomalies such as misassigned capitals, geographic
centroids, duplicate coordinates, GBIF headquarters artifacts,
institution-proximate records, distributional outliers, marine points in
terrestrial datasets, and zero/placeholder coordinates.</p>
<p>Then each validated record is matched to its corresponding WGSRPD
level-3 Botanical Country classifications with different status
(native/introduced/extinct/location_doubtful), which minimizes
mis-classification risks in cross-regional analyses.</p>
<p>To optimize computational efficiency, this module supports
parallelization through multi-threading by specifying the desired
concurrency level via the <code>threads</code> parameter.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">refine</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/refine_records.html">refine_records</a></span><span class="op">(</span>voucher <span class="op">=</span> <span class="va">voucher</span>,</span>
<span>                            threads <span class="op">=</span> <span class="fl">4</span>,</span>
<span>                            save_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">dirname</a></span> <span class="op">(</span><span class="va">gbif_occurrence_file</span><span class="op">)</span>,</span>
<span>                            tests <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"capitals"</span>, <span class="st">"centroids"</span>, <span class="st">"equal"</span>, <span class="st">"gbif"</span>, <span class="st">"institutions"</span>, <span class="st">"outliers"</span>, <span class="st">"seas"</span>,<span class="st">"zeros"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><code>refine</code> includes <code>all_records</code> for all cleaned
records and <code>native_records</code> for native of them, and time
costs. If you set <code>save_path</code>, then <code>all_records</code>
and <code>native_records</code> are exported, named as
<code>usable_refined_records.csv.gz</code> and
<code>native_refined_records.csv.gz</code>, both are potentially
meritorious for your future research.</p>
</div>
<div class="section level2">
<h2 id="map-records">Map records<a class="anchor" aria-label="anchor" href="#map-records"></a>
</h2>
<p>This module is implemented optionally, rendering the output from
<code>coordinate_refine</code>, employing <code>geohashTools</code><span class="citation">(Chirico 2023)</span> to perform spatial clustering of
geo-coordinates. The <code>precision</code> parameter determines grid
resolution (e.g. 4 for 20 km×20 km grids, 3 for 156 km×156 km grids). It
groups records by <code>UltraGBIF_wcvp_taxon_name</code>, WGSRPD
information and <code>geohash grid</code> and follows deduplication.</p>
<p>After simplification, dynamic maps with different layers are rendered
by mapview<span class="citation">(Appelhans et al. 2023)</span> with
WGSRPD information legend. Symbol size is controlled by the
<code>cex</code> parameter. Moreover, they supports zoom functionality
for spatial resolution adjustment and enables interactive selection of
individual coordinates to display their full attributes.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/map_records.html">map_records</a></span><span class="op">(</span>records_refined <span class="op">=</span> <span class="va">refine</span>,</span>
<span>                     precision <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                     cex <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-mapview" class="csl-entry">
Appelhans, Tim, Florian Detsch, Christoph Reudenbach, and Stefan
Woellauer. 2023. <span>“Mapview: Interactive Viewing of Spatial Data in
r.”</span> <a href="https://CRAN.R-project.org/package=mapview" class="external-link">https://CRAN.R-project.org/package=mapview</a>.
</div>
<div id="ref-boyle2013" class="csl-entry">
Boyle, Brad, Nicole Hopkins, Zhenyuan Lu, Juan Antonio Raygoza Garay,
Dmitry Mozzherin, Tony Rees, Naim Matasci, et al. 2013. <span>“The
Taxonomic Name Resolution Service: An Online Tool for Automated
Standardization of Plant Names.”</span> <em>BMC Bioinformatics</em> 14
(1): 16. <a href="https://doi.org/10.1186/1471-2105-14-16" class="external-link">https://doi.org/10.1186/1471-2105-14-16</a>.
</div>
<div id="ref-geohashTools" class="csl-entry">
Chirico, Michael. 2023. <span>“geohashTools: Tools for Working with
Geohashes.”</span> <a href="https://CRAN.R-project.org/package=geohashTools" class="external-link">https://CRAN.R-project.org/package=geohashTools</a>.
</div>
<div id="ref-demelo2024" class="csl-entry">
De Melo, Pablo Hendrigo Alves, Nadia Bystriakova, Eve Lucas, and
Alexandre K. Monro. 2024. <span>“A New R Package to Parse Plant Species
Occurrence Records into Unique Collection Events Efficiently Reduces
Data Redundancy.”</span> <em>Scientific Reports</em> 14 (1): 5450. <a href="https://doi.org/10.1038/s41598-024-56158-3" class="external-link">https://doi.org/10.1038/s41598-024-56158-3</a>.
</div>
<div id="ref-govaerts2021" class="csl-entry">
Govaerts, Rafaël, Eimear Nic Lughadha, Nicholas Black, Robert Turner,
and Alan Paton. 2021. <span>“The World Checklist of Vascular Plants, a
Continuously Updated Resource for Exploring Global Plant
Diversity.”</span> <em>Scientific Data</em> 8 (1): 215. <a href="https://doi.org/10.1038/s41597-021-00997-6" class="external-link">https://doi.org/10.1038/s41597-021-00997-6</a>.
</div>
<div id="ref-zizka2019" class="csl-entry">
Zizka, Alexander, Daniele Silvestro, Tobias Andermann, Josué Azevedo,
Camila Duarte Ritter, Daniel Edler, Harith Farooq, et al. 2019.
<span>“CoordinateCleaner : Standardized Cleaning of Occurrence Records
from Biological Collection Databases.”</span> Edited by Tiago Quental.
<em>Methods in Ecology and Evolution</em> 10 (5): 744–51. <a href="https://doi.org/10.1111/2041-210X.13152" class="external-link">https://doi.org/10.1111/2041-210X.13152</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Yuxuan Wang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
